File: iOS/PhenixMultiAngleOnDemand/PhenixCore/Stream/Act.swift
//
//  Copyright 2021 Phenix Real Time Solutions, Inc. Confidential and Proprietary. All rights reserved.
//

import Foundation

public struct Act: CustomStringConvertible {
    public var title: String
    public var offset: TimeInterval

    public var description: String {
        "Act, title: \(title), offset: \(offset)"
    }

    public init?(time: String) {
        let result = time.split(separator: ":")

        guard let minutes = TimeInterval(result[0]) else {
            return nil
        }

        guard let seconds = TimeInterval(result[1]) else {
            return nil
        }

        let timeinterval: TimeInterval = minutes * 60 + seconds

        self.title = time
        self.offset = timeinterval
    }
}

public extension Act {
    // swiftlint:disable force_unwrapping
    static let zero = Act(time: "00:00")!
}

File: Android/PhenixMultiAngleOnDemand/app/src/main/java/com/phenixrts/suite/phenixmultiangleondemand/models/Act.kt
/*
 * Copyright 2021 Phenix Real Time Solutions, Inc. Confidential and Proprietary. All rights reserved.
 */

package com.phenixrts.suite.phenixmultiangleondemand.models

import kotlinx.serialization.Serializable
import kotlinx.serialization.decodeFromString
import kotlinx.serialization.encodeToString
import kotlinx.serialization.json.Json

@Serializable
data class Act(val title: String, val offsetFromBeginning: Long)

fun String.asAct(): Act? = try {
    Json{ ignoreUnknownKeys = true }.decodeFromString(this)
} catch (e: Exception) {
    null
}

fun Act.toJson(): String = Json.encodeToString(this)
